src/transaction.cc

```diff
@@ -1700,7 +1700,7 @@ std::string Transaction::toJSON(int parts) {
             reinterpret_cast<const unsigned char*>("messages"),
             strlen("messages"));
         yajl_gen_array_open(g);
-        for (auto a : m_rulesMessages) {
+        for (auto& a : m_rulesMessages) {
```

Comment:
```suggestion
        for (const auto& a : m_rulesMessages) {
```
Probably here you could use `const` too.